props/prop_base.h

```diff
@@ -734,6 +741,7 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 
     ONCEPERBLADE(DEACTIVATE);
     SaveVolumeIfNeeded();
+    FindBestConfig(true);
```

Comment:
Is the only reason you need this for the printouts?
Is the only reason you need those printouts because of the change in FindBlade() ?

Reply:
Originally,FindBlade() called SetPreset(0, false), so there was no printout of current preset and no call to DoNewFont(). There was no indication something happened when a blade was inserted.
I thought playing the font.wav made sense to know something happened, and to indicate what preset was loaded, so I changed it to SetPreset(0, true). However, that caused 2 boot.wavs to play simultaneously. I don't know why SetPreset would play boot.wav. But as a workaround, I added the SetPreset(0, false) back in, but in the condition of only the first 3 seconds so a second boot.wav wasn't triggered.

The code currently works as I stated above, which was the intent. Is it bad?

> Calling id(true) will print (and speak) ID, else quiet.
> FindBladeAgain() is announced along with the SetPreset routine being run. (except for first 3 seconds of boot to avoid 2 boot.wavs being played)
> FindBestConfig() runs quietly otherwise.
> "scanid" calls FindBladeAgain(),
> "id" calls id(true) directly, so just an announce.
> If BLADE_ID_SCAN_MILLIS is not defined, "scanid" detects the new blade ID.

Reply:
> Originally,FindBlade() called SetPreset(0, false), so there was no printout of current preset and no call to DoNewFont(). There was no indication something happened when a blade was inserted.
> I thought playing the font.wav made sense to know something happened, and to indicate what preset was loaded, so I changed it to SetPreset(0, true). However, that caused 2 boot.wavs to play simultaneously.

So far so good.

> I don't know why SetPreset would play boot.wav. But as a workaround, I added the SetPreset(0, false) back in, but in the condition of only the first 3 seconds so a second boot.wav wasn't triggered.

This is bad.
1. it's always easier to fix a problem you understand
2. functions should not do different things depending on what time it is, that makes them hard to use and unpredictable. In this particular case, it will lead to people who use a delay during boot to hear boot sounds twice.
3. Maybe it would be better to fix the *other* place that plays a boot sound?

So, good intentions, bad implementation.

Reply:
🤦‍♂️
It's not 2 boots. It's boot.wav and font.wav. I had the same sound for both in a Dooku font.
Lesson learned, always use my testfont.
Well at least that makes more sense.
Ok, working on it.

Reply:
ok fixed.